internal/alpini/netty4/alpini-netty4-base/src/main/java/com/linkedin/alpini/netty4/ssl/SslInitializer.java

```diff
@@ -370,6 +379,11 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
       if (!ctx.isRemoved()) {
         next(ctx);
       }
+      if (ctx.channel().hasAttr(SSL_HANDSHAKE_START_TS)) {
+        // If SSL_HANDSHAKE_START_TS is set, then the handshake was started.
+        // If the connection closes mid-handshake, userEventTriggered might never fire.
+        _handshakesFailed.increment();
```

Comment:
QQ, do we clear this attribute in `userEventTriggered`?
`ctx.channel().attr(SSL_HANDSHAKE_START_TS).getAndSet(null)`

Just a heads-up, Netty [doc](https://netty.io/4.0/api/io/netty/util/Attribute.html#getAndRemove--) seems to say that `getAndSet(null)` might still keep the attribute in the AttribyteMap, so hasAttr(SSL_HANDSHAKE_START_TS) still returns true? Wondering if this could actually be an issue or not. 
> If you only want to return the old value and clear the [Attribute](https://netty.io/4.0/api/io/netty/util/Attribute.html) while still keep it in the [AttributeMap](https://netty.io/4.0/api/io/netty/util/AttributeMap.html) use [getAndSet(Object)](https://netty.io/4.0/api/io/netty/util/Attribute.html#getAndSet-T-) with a value of null.

Reply:
Yes! Good catch. I believe it would be an issue.
If the channel becomes inactive after SSL handshakes completed exceptionally, we would end up incrementing the failed counter twice.